### PR TITLE
[Doc]Update docs for OAP 0.8.2 release

### PR DESCRIPTION
**Pull request by HongW2019**
## How was this patch tested?

### CODE REVIEW

File: oap-cache/oap/docs/User-Guide.md [hunk 1]
```diff
@@ -7,7 +7,7 @@
 * [Working with SQL Index](#Working-with-SQL-Index)
 * [Working with SQL Data Source Cache](#Working-with-SQL-Data-Source-Cache)
 * [Run TPC-DS Benchmark](#Run-TPC-DS-Benchmark)
-* [Advanced Configurations Scenarios](#Advanced-Configurations-Scenarios])
+* [Configuration for Advanced Scenarios](#Configuration-for-Advanced-Scenarios)
```
Comment (HongW2019) [inline oap-cache/oap/docs/User-Guide.md:10]: @jerrychenhf Please help review this part's title and structure.

File: dev/prepare_oap_env.sh [hunk 2]
```diff
@@ -144,7 +144,7 @@ function prepare_memkind() {
   fi
   cd memkind/
   git pull
-  git checkout v1.10.0-oap-0.7
+  git checkout v1.10.1-rc2
```
Comment (yao531441) [inline dev/prepare_oap_env.sh:147]: Thanks, Hong.LGTM!

File: oap-cache/oap/docs/User-Guide.md (hunk 1 above)
Comment (zhixingheyi-tian) [reply oap-cache/oap/docs/User-Guide.md:10]: Advanced Configurations

Comment (HongW2019) [reply oap-cache/oap/docs/User-Guide.md]: Modified.

**Comment by HongW2019** [comment]
@yao531441 Please review developer & installation docs and corresponding scripts.

**Pull request merged**